actionview/lib/action_view/base.rb

```diff
@@ -205,7 +205,8 @@ def changed?(other) # :nodoc:
     delegate :formats, :formats=, :locale, :locale=, :view_paths, :view_paths=, to: :lookup_context
 
     def assign(new_assigns) # :nodoc:
-      @_assigns = new_assigns.each { |key, value| instance_variable_set("@#{key}", value) }
+      @_assigns = new_assigns
+      new_assigns.each { |key, value| instance_variable_set("@#{key}", value) }
```

Comment:
I think `ActionView::Base` will inevitably be marked has having too many shapes, but at least we can keep a common root...

cc @tenderlove what do you think?